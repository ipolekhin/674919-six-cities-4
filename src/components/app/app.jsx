import React from "react";
import {BrowserRouter, Route, Switch} from "react-router-dom";
import PageContainer from "../page-container/page-container.jsx";
import Header from "../header/header.jsx";
import Main from "../main/main.jsx";
import Offer from "../offer/offer.jsx";
import {countPlacesType, placeCardsType} from "../../types/types";
import {TownCoordinates, TownType} from "../../const";

export default class App extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {offer: null};
    this.onTitleClick = this.onTitleClick.bind(this);
  }

  render() {
    const {placeCards} = this.props;

    return (
      <React.Fragment>
        <PageContainer>
          <Header/>
          <BrowserRouter>
            <Switch>
              <Route exact path="/">
                {this._renderOfferScreen()}
              </Route>
              <Route exact path="/dev-offer">
                <Offer
                  cityCoordinate = {TownCoordinates[TownType.AMSTERDAM]}
                  offer = {placeCards[0]}
                  placeCards = {placeCards}
                  onTitleClick = {this.onTitleClick}
                />
              </Route>
            </Switch>
          </BrowserRouter>
        </PageContainer>
      </React.Fragment>
    );
  }

  _renderOfferScreen() {
    const {countPlaces, placeCards} = this.props;
    if (this.state.offer === null) {
      return (
        <Main
          countPlaces = {countPlaces}
          placeCards = {placeCards}
          onTitleClick = {this.onTitleClick}
          cityCoordinate = {TownCoordinates[TownType.AMSTERDAM]}
        />
      );
    } else {
      return (
        <Offer
          offer = {this.state.offer}
        />
      );
    }

  }

  onTitleClick(offerId) {
    this.setState({offer: this.props.placeCards.find((card) => card.id === offerId)});
  }
}

App.propTypes = {
  countPlaces: countPlacesType,
  placeCards: placeCardsType,
};
