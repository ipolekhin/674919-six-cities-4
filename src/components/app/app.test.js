import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app.jsx";

const currentCity = `Amsterdam`;

const offers = [{
  adults: 3,
  townName: `Amsterdam`,
  bedrooms: 2,
  cardName: `Wood and stone place`,
  cardRating: 4.6,
  cardRatingStars: `92%`,
  cardType: `Room`,
  coordinatesItem: [52.3909553943508, 4.85309666406198],
  description: `Text text....`,
  id: `1`,
  image: `img/apartment-01.jpg`,
  images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
  insideItems: [`wi-fi`, `bathroom`],
  premiumPlace: true,
  price: 100,
  reviews: [
    {
      date: new Date(2020, 7, 17),
      id: `01`,
      rating: `92%`,
      text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      userAvatar: `img/avatar-max.jpg`,
      userName: `Max`,
    }
  ],
  user: {
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    pro: true,
  },
}];

const onCityClick = () => {};

it(`Render App`, () => {
  const tree = renderer
    .create(
        <App
          currentCity = {currentCity}
          offers = {offers}
          onCityClick = {onCityClick}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
