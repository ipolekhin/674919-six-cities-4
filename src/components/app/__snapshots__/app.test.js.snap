// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Render App 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Wood and stone place
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "92%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.6
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              3
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              100
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                wi-fi
              </li>
              <li
                className="property__inside-item"
              >
                bathroom
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Text text....
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              className="reviews__list"
            />
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        />
      </section>
    </div>
  </main>
</div>
`;
